verification/verification_test.go

```diff
@@ -250,3 +250,69 @@ func testCertifyKey(d TestDPEInstance, t *testing.T) {
 
 	// TODO: When DeriveChild is implemented, call it here to add more TCIs and call CertifyKey again.
 }
+
+func TestTagTCI(t *testing.T) {
+	s := &DpeSimulator{exe_path: *sim_exe, supports: Support{AutoInit: true, Tagging: true}}
+	s.SetLocality(DPE_SIMULATOR_AUTO_INIT_LOCALITY)
+	if s.HasPowerControl() {
+		err := s.PowerOn()
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer s.PowerOff()
+	}
+
+	client, err := NewClient256(s)
+	if err != nil {
+		t.Fatalf("Could not initialize client: %v", err)
+	}
+
+	// Try to create the default context if isn't done automatically.
+	if !s.GetSupport().AutoInit {
+		initCtxResp, err := client.InitializeContext(NewInitCtxIsDefault())
+		if err != nil {
+			t.Fatalf("Failed to initialize default context: %v", err)
+		}
+		defer client.DestroyContext(NewDestroyCtx(initCtxResp.Handle, false))
+	}
+
+	tag := TCITag(12345)
+	// Check to see our tag is not yet found.
+	if _, err := client.GetTaggedTCI(&GetTaggedTCIReq{Tag: tag}); !errors.Is(err, StatusBadTag) {
+		t.Fatalf("GetTaggedTCI returned %v, want %v", err, StatusBadTag)
+	}
+
+	// Tag the default context
+	var ctx ContextHandle
+
+	tagResp, err := client.TagTCI(&TagTCIReq{ContextHandle: ctx, Tag: tag})
+	if err != nil {
+		t.Fatalf("Could not tag TCI: %v", err)
+	}
+
+	if tagResp.NewContextHandle != ctx {
+		t.Errorf("New context handle from TagTCI was %x, expected %x", tagResp.NewContextHandle, ctx)
+	}
```

Comment:
This is only the expected behavior because the input handle is the default context. It might be good to

* Add a comment stating this
* (optional) Add a check that if the input handle is a random handle, the output is a different random value. But this might make sense as a separate test case that just checks the handles for all commands. Not sure.

Reply:
I was going to say it would be good to add a simulation context to run some extra tests against, but it doesn't really make sense without extending the simulation context's TCI before doing any tests. This sounds like it would be better in a separate PR.

Reply:
Could this be folded into the TODO at the bottom of this test? Is it really a second TODO?

Reply:
For my comment, it falls in the TODO. So no action for me.